kale/loaddata/image_access.py

```diff
@@ -336,42 +337,89 @@ def get_cifar(cfg):
     return train_loader, valid_loader
 
 
-def read_dicom_images(dicom_path, sort_instance=True, sort_patient=False):
-    """Read dicom images for multiple patients and multiple instances/phases.
+def read_dicom_phases(dicom_path, sort_instance=True):
+    """Read dicom images of multiple instances/phases for one patient.
 
     Args:
         dicom_path (str): Path to DICOM images.
+        sort_instance (bool, optional): Whether sort images by InstanceNumber (i.e. phase number). Defaults to True.
+
+    Returns:
+        [list]: List of dicom phases
+    """
+    ds = []
+    phase_files = glob.glob(dicom_path + "/**/*.dcm", recursive=True)
+    for phase_file in phase_files:
+        dataset = pydicom.dcmread(phase_file)
+        ds.append(dataset)
+    if sort_instance:
+        ds.sort(key=lambda x: x.InstanceNumber, reverse=False)
+
+    return ds
+
+
+def read_dicom_dir(dicom_path, sort_instance=True, sort_patient=False):
+    """Read dicom files for multiple patients and multiple instances / phases from a given directory arranged in the
+        following structure: ::
+
+            root/patient_a/.../phase_1.dcm
+            root/patient_a/.../phase_2.dcm
+            root/patient_a/.../phase_3.dcm
+
+            root/patient_b/.../phase_1.dcm
+            root/patient_b/.../phase_2.dcm
+            root/patient_b/.../phase_3.dcm
+
+            root/patient_m/.../phase_1.dcm
+            root/patient_m/.../phase_2.dcm
+            root/patient_m/.../phase_3.dcm
+
+    Args:
+        dicom_path (str): Directory of DICOM files.
         sort_instance (bool, optional): Whether sort images by InstanceNumber (i.e. phase number) for each subject.
             Defaults to True.
         sort_patient (bool, optional): Whether sort subjects' images by PatientID. Defaults to False.
 
     Returns:
-        [array-like]: [description]
+        [list[list]]: [a list of dicom phase lists]
     """
-    sub_dirs = os.listdir(dicom_path)
-    all_ds = []
-    sub_ids = []
+    sub_dirs = [os.path.join(dicom_path, sub_dir) for sub_dir in os.listdir(dicom_path)]
+    sub_dirs = filter(os.path.isdir, sub_dirs)
+    ds = []
```

Comment:
ds is too short be be readable.

Reply:
I followed the variable name in [pydicom](https://pydicom.github.io/pydicom/stable/tutorials/dataset_basics.html). Some clincians (like Sharkey) also use `ds` for dicom dataset objects.

Reply:
I think it will be better to follow conventions in *PyTorch* rather than PyDicom. See how PyTorch name dataset variables and how we name other dataset variables in pykale (if any).